### ISSUE TITLE
AddThirdPartyMojo, unable to define missing licenses for specific dependencies

### ISSUE THREAD

**Issue by rez-** [mojohaus/license-maven-plugin#174]
When using a missing file for the missing licenses, the following two licenses are always missing.

groovy--groovy-all-1.0-jsr--04=asdf
nanocontainer-remoting--nanocontainer-remoting--1.0-RC-1=asdf

This is the log:
```
[INFO] Load missing file .../THIRD-PARTY.properties
[WARNING] dependency [groovy--groovy-all-1.0-jsr--04] does not exist in project, remote it from the missing file.
[WARNING] dependency [nanocontainer-remoting--nanocontainer-remoting--1.0-RC-1] does not exist in project, remote it from the missing file.
[INFO] Missing file .../THIRD-PARTY.properties is up-to-date.
[WARNING] There is 2 dependencies with no license :
[WARNING]  - groovy--groovy-all-1.0-jsr--04
[WARNING]  - nanocontainer-remoting--nanocontainer-remoting--1.0-RC-1
[INFO] Writing third-party file to .../THIRD-PARTY.txt
[INFO] Regenerate missing license file .../THIRD-PARTY.properties
```

**Comment by rez-** [comment]
Also note that it first detects the dependencies correctly, but afterwards incorrectly stores the groupId for nanocontainer-remoting, and the artifactId and version for groovy-all:
```
...
[INFO] detected artifact **nanocontainer**--nanocontainer-remoting--1.0-RC-1
[INFO] add dependency [**nanocontainer**--nanocontainer-remoting--1.0-RC-1]
...
[INFO] detected artifact groovy--**groovy-all**--1.0-jsr-04
[INFO] add dependency [groovy--**groovy-all**--1.0-jsr-04]
...
```

**Comment by rez-** [comment]
When I correctly store the dependencies in the missingFile, the missing licenses wont be removed, but the THIRD-PARTY.txt file has the incorrect groupId/artifactId/version.

THIRD-PARTY.properties:
groovy--groovy-all--1.0-jsr-04
nanocontainer--nanocontainer-remoting--1.0-RC-1=asdf

THIRD-PARTY.txt:
(asdf) groovy-all-1.0-jsr (groovy:groovy-all-1.0-jsr:04 - no url defined)
(asdf) Nanocontainer Remoting (nanocontainer-remoting:nanocontainer-remoting:1.0-RC-1 - no url defined)

**Comment by ppalaga** [comment]
@rez- could you please reproduce with a snapshot built from the current master? 
If it is still reproducible, an integration test in https://github.com/mojohaus/license-maven-plugin/tree/master/src/it would be highly welcome.

**Comment by rez-** [comment]
Unfortunately it's still reproducible, so I just created a pull request (#183).

**Comment by rez-** [comment]
After some digging, it turns out that the nanocontainer-remoting artifact is stored under groupId "nanocontainer" but the pom file contains the groupId "nanocontainer-remoting"...

**Comment by rez-** [comment]
Same with groovy-all,
The pom file has artifactId "groovy-all-1.0-jsr" with version "04"
But maven central lists the artifactId as groovy-all and the version as "1.0-jsr-04"

**Comment by rez-** [comment]
see MavenProjectBuilder.buildFromRepository in DefaultDependenciesTool.java

**Comment by ppalaga** [comment]
> nanocontainer-remoting artifact is stored in maven central under the groupId "nanocontainer" but the pom file contains the groupId "nanocontainer-remoting"

> The pom file has artifactId "groovy-all-1.0-jsr" with version "04"
> But maven central lists the artifactId as groovy-all and the version as "1.0-jsr-04"

Interesting. Thanks for looking into it. Looks like it has been reported to Sonatype, but they did not want to fix it:

>  there isn't anything I can do about the bad POM files. It's Sonatype policy not to touch artifact files once they've been deployed to Central. https://issues.sonatype.org/browse/MVNCENTRAL-209

I think we should try to workaround it.

**Comment by rez-** [comment]
I have half a work-around in commit 6dc3b2f, the missing dependency listing now shows it correct:
```
[DEBUG] There are 2 dependencies with no license from poms :
[DEBUG] - nanocontainer--nanocontainer-remoting--1.0-RC-1
[DEBUG] - groovy--groovy-all--1.0-jsr-04
```
However, the values are still not correct in the THIRD-PARTY.txt file

**Comment by rez-** [comment]
Fixed in #190 :D
